fastestimator/pipeline.py

```diff
@@ -139,102 +139,111 @@ def get_per_epoch(ops: Iterable[Union[T, Scheduler[T]]], epoch: int) -> List[T]:
+        eval_data (fe.data.Dataset): A dataset for evaluation.
         batch_size (int, Scheduler): The batch size to use during training
         ops (list, fe.op): A list of operations to be applied within the data pipeline
-        drop_last (bool): Whether to drop the last batch in an epoch if there aren't enough remaining elements for a
-                        complete batch
-        num_process (int): How many CPUs to use in the pipeline. None will auto-select based on performance tests. 
+        num_process (int): How many CPUs to use in the pipeline. None will auto-select based on performance tests.
                             You might need to set this to zero if you want to use debuggers
-        shuffle_train (bool): Whether to shuffle the training data
-        shuffle_eval (bool): Whether to shuffle the eval data
     """
     def __init__(self,
-                 train_data: Optional[Dataset] = None,
-                 eval_data: Optional[Dataset] = None,
-                 batch_size: Union[int, Scheduler] = 1,
+                 train_data: Union[Dataset, Scheduler],
+                 eval_data: Union[Dataset, Scheduler, None] = None,
+                 batch_size: Union[int, Scheduler, None] = None,
                  ops: Union[None, NumpyOp, Scheduler[NumpyOp], Iterable[Union[NumpyOp, Scheduler[NumpyOp]]]] = None,
-                 drop_last: bool = False,
-                 num_process: Optional[Union[int, Scheduler[int]]] = None,
-                 shuffle_train: bool = True,
-                 shuffle_eval: bool = False):
+                 num_process: Union[int, None] = None):
+        super().__init__(train_data=train_data, eval_data=eval_data)
         self.ops = [] if ops is None else to_list(ops)
-        if isinstance(batch_size, Scheduler):
-            assert 0 in batch_size.epoch_dict.keys(), "Batch size must be specified for epoch 0"
-            assert all(map(lambda x: x is not None, batch_size.epoch_dict.values())), "Batch size must never be None"
-        else:
-            batch_size = Scheduler({0: batch_size})
         self.batch_size = batch_size
-        self.drop_last = drop_last
-        self.num_process = os.cpu_count() or 1 if num_process is None else num_process
+        self.num_process = os.cpu_count() if num_process is None else num_process
         self.datasets = {}
         self.dataloaders = {}
-        if train_data is not None:
-            self.datasets["train"] = self._build_dataset(train_data, "train")
-            # Intentionally not using self.num_process in the build_loader call since want to trigger perf test
-            self.dataloaders["train"] = self._build_loader(self.datasets["train"],
-                                                           shuffle=shuffle_train,
-                                                           num_process=num_process)
+        self.datasets["train"] = self._build_dataset(train_data, "train")
```

Comment:
train_data should be optional so that as a user with a trained model i can still instantiate a pipeline only for testing or validation data